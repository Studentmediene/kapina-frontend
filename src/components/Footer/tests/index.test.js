import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from '../';

describe('<Footer />', () => {
  it('renders correctly', () => {
    const props = {
      chiefEditor: 'chiefEditor',
      radioEditor: 'radioEditor',
      loading: false,
      error: false,
      loadFooter: jest.fn(),
    };

    const tree = shallow(<Footer {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly while loading', () => {
    const props = {
      chiefEditor: 'chiefEditor',
      radioEditor: 'radioEditor',
      loading: true,
      error: false,
      loadFooter: jest.fn(),
    };

    const tree = shallow(<Footer {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly on error', () => {
    const props = {
      chiefEditor: 'chiefEditor',
      radioEditor: 'radioEditor',
      loading: false,
      error: true,
      loadFooter: jest.fn(),
    };

    const tree = shallow(<Footer {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
