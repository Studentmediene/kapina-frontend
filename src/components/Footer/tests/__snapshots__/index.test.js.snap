// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<div
  className="footer"
>
  <div
    className="content"
  >
    <div>
      Denne tjenesten tilbys av Studentmediene i Trondheim AS. Musikken er gjengitt med tilatelse fra TONO/NCB.
    </div>
    <div>
      Uautorisert lenking, videreføring eller kopiering er ulovlig.
    </div>
    <br />
    <div
      className="bold"
    >
      Kontakt oss
    </div>
    <a
      className="footerLink"
      href="mailto:redaktor@radiorevolt.no"
    >
      Radioredaktør: 
      radioEditor
    </a>
    <a
      className="footerLink"
      href="mailto:ansvarligredaktor@studentmediene.no"
    >
      Ansvarlig redaktør: 
      chiefEditor
    </a>
    <a
      className="footerLink"
      href="mailto:musikkprodusent@studentmediene.no"
    >
      Musikkprodusent: 
      musicProducer
    </a>
    <br />
    <div
      className="socialMediaLinks"
    >
      <SocialMediaContainer />
      <br />
    </div>
    <Link
      className="footerLink"
      replace={false}
      to="/om"
    >
      Om oss
    </Link>
    <Link
      className="footerLink"
      replace={false}
      to="/personvern"
    >
      Personvern
    </Link>
    <p>
      2019
       © Radio Revolt
    </p>
  </div>
</div>
`;

exports[`<Footer /> renders correctly on error 1`] = `
<div
  className="footer"
>
  <div
    className="content"
  >
    <div>
      Denne tjenesten tilbys av Studentmediene i Trondheim AS. Musikken er gjengitt med tilatelse fra TONO/NCB.
    </div>
    <div>
      Uautorisert lenking, videreføring eller kopiering er ulovlig.
    </div>
    <br />
    <div
      className="bold"
    >
      Kontakt oss
    </div>
    <a
      className="footerLink"
      href="mailto:redaktor@radiorevolt.no"
    >
      Radioredaktør: 
      radioEditor
    </a>
    <a
      className="footerLink"
      href="mailto:ansvarligredaktor@studentmediene.no"
    >
      Ansvarlig redaktør: 
      chiefEditor
    </a>
    <a
      className="footerLink"
      href="mailto:musikkprodusent@studentmediene.no"
    >
      Musikkprodusent: 
      musicProducer
    </a>
    <br />
    <div
      className="socialMediaLinks"
    >
      <SocialMediaContainer />
      <br />
    </div>
    <Link
      className="footerLink"
      replace={false}
      to="/om"
    >
      Om oss
    </Link>
    <Link
      className="footerLink"
      replace={false}
      to="/personvern"
    >
      Personvern
    </Link>
    <p>
      2019
       © Radio Revolt
    </p>
  </div>
</div>
`;

exports[`<Footer /> renders correctly while loading 1`] = `
<div
  className="footer"
>
  <div
    className="content"
  >
    <div>
      Denne tjenesten tilbys av Studentmediene i Trondheim AS. Musikken er gjengitt med tilatelse fra TONO/NCB.
    </div>
    <div>
      Uautorisert lenking, videreføring eller kopiering er ulovlig.
    </div>
    <br />
    <div
      className="bold"
    >
      Kontakt oss
    </div>
    <a
      className="footerLink"
      href="mailto:redaktor@radiorevolt.no"
    >
      Radioredaktør: 
      radioEditor
    </a>
    <a
      className="footerLink"
      href="mailto:ansvarligredaktor@studentmediene.no"
    >
      Ansvarlig redaktør: 
      chiefEditor
    </a>
    <a
      className="footerLink"
      href="mailto:musikkprodusent@studentmediene.no"
    >
      Musikkprodusent: 
      musicProducer
    </a>
    <br />
    <div
      className="socialMediaLinks"
    >
      <SocialMediaContainer />
      <br />
    </div>
    <Link
      className="footerLink"
      replace={false}
      to="/om"
    >
      Om oss
    </Link>
    <Link
      className="footerLink"
      replace={false}
      to="/personvern"
    >
      Personvern
    </Link>
    <p>
      2019
       © Radio Revolt
    </p>
  </div>
</div>
`;
